src/compiler/core.ts

```diff
@@ -1399,6 +1399,32 @@ namespace ts {
     /** Returns lower case string */
     export function toLowerCase(x: string) { return x.toLowerCase(); }
 
+    const fileNameLowerCaseRegExp = /[^İi̇ıßa-z0-9\\/:\-_\. ]+/g;

```

Comment:
@DanielRosenwasser may have more opinions here (since he brought it up the last time I tried to add one in a PR), but ostensibly including Unicode characters in our source has the potential to slow down parsing our source (which is very annoying when Unicode characters are appropriate in comments), so it may be appropriate to remove the characters from the comments and use hexadecimal character references in the regex.

Reply:
Although I note we've probably already blown this in `core.ts`, so it's probably best to enforce via a lint rule if we actually think it's a problem.

@DanielRosenwasser  ?